### DISCUSSION TITLE
Proxy not working in Nuxt 3

### DISCUSSION THREAD

**Discussion by lq9958** [nuxt/nuxt#25018] [Questions] [answered]
I am using a proxy to access the backend and fetch data in Nuxt 3, but the console keeps showing the error:
```powershell
 [Vue Router warn]: No match found for location with path "/cms/page/queryPages"
 ```
here is my proxy configuration
```json
{
  nitro: {
    devProxy: {
      "/cms": {
        target: "http://localhost:4090",
        changeOrigin: true,
      },
      "/public": {
        target: "http://localhost:4090",
        changeOrigin: true,
      },
    },
  },
}
```

This is how I use $fetch to send requests in my business logic:
```javascript
<script setup>
const { data: pageData, refresh } = await useAsyncData(async () => {
  const pageDataResponse = await $fetch("/cms/page/queryPages", {
    method: "POST",
  });
})
console.log(pageData.value)
</script>
```
but when I print pageData, the browser console shows `null`. It remains null even when printed in the onMounted lifecycle. However, if I set a timer and print after 1 second, it has a value. So, I currently have two questions: 
* First, why does the vue-router console report an error?
* Second, isn't useAsyncData executed on the server side already? Why is the data null in the browser?

**Comment by xjccc** [comment]
emm..
the docs show you how to use it

try this to use useAsyncData?

[useAsyncData](https://nuxt.com/docs/api/composables/use-async-data#usage)

nuxt in browser can show your console in `setup` at server side

**Reply by lq9958** [reply]
Thank you for your response, but my concerns are not only related to `useAsyncData`. Why does the vue-router console keep warning? Those addresses are just my API endpoints. Also, shouldn't `useAsyncData` be executed on the server side? Why is the value null and requires a `refresh` to get data? What's the difference with client-side rendering? It's really confusing.

**Reply by xjccc** [reply]
I dont know why router console warning in your case.

As usual NuxtLink href could find the path from your router.

You can provide a reproduction.

I think following the docs you can fetch data correctly at server side.

**Comment by davidstackio** [comment, marked as answer]
Check out this video from one of the maintainers of Nuxt. It helped me get a better understanding of the different methods of proxying in Nuxt.

https://www.youtube.com/watch?v=J4E5uYz5AY8

**Reply by lq9958** [reply]
Oh my God! You know, I have been working on this problem for three days. it's worked!!! amazing!! thank you sooooo much!!

**Reply by TheAlexLichter** [reply]
Thanks for sharing my video @davidstackio 🙏🏻
Glad it worked @lq9958 ☺️
